### DISCUSSION TITLE
Woche 4.12-11.12.23

### DISCUSSION THREAD

**Discussion by carolina-cza** [carolina-cza/Tecna#3] [Weekly Blog]
Die Woche haben wir eine kurze Einführung in das Thema Programmierung bekommen und schon selbstständig ein paar Codes in Python geschrieben.
Heute haben wir uns vor allem mit dem Artikel "Energy Efficiency across Programming Languages" beschäfftigt. Dafür haben wir uns jeweils in zweier und dreier Gruppen zusammen gefunden und jede Gruppe hat für sich dann eine Aufgabe zu einer bestimmten Section des Artikels bekommen, welches sie dann bearbeiten mussten. Anschließend hat dann jede Gruppe seine Ergebnisse präsentiert, sodass alle auf dem selben Stand sind.
Unsere Zusammenschrift:
### Section 2.3 Gruppe 4

**1.      Können wir das Experiment in kleinem Maßstab replizieren?**

_a.      Ja, die Daten der CLBGs (Computing Language Benchmark Games) stehen online mit genauen Experimentdaten zur Verfügung. 
          Dadurch kann man, wie in der Fachschrift beschrieben, das Experiment selbst wiederholen._

**2.      Können wir eine allgemeine Aussage zu (c)(i)(v) treffen:**

_a.      c=compiled
         Übersetzt Programmiersprache in eine Form, die der Computer ausführen kann
b.      i=interpreted
        Übersetzt Programmiersprache direkt in Maschinencode, der vom Computer ausgeführt wird
c.      v=virtual-machine language
         Code läuft unabhängig von der Hardware
d.      Generell sind Sprachen, die kompilierbar sind in den Kategorien die energieeffizientesten, Sprachen, die virtual-machine_ 
         basierende Sprachen sind eher in der Mitte und interpreted Sprachen haben hauptsächlich hohe Energieintensität

**3.**      
_Außerdem kann man aussagen, dass Sprachen, die eher energieintensiv sind, weniger Peak Memory haben._

**Comment by ElinaGojani** [comment]
Sehr gut zusammengefasst. Weiter so :)
